### PR TITLE
Remove dynamic account behaviors.

### PR DESCRIPTION
**Pull request by derekcollison**
We used these in tests and for experimenting with sandboxed environments like the demo network.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/route.go [hunk 1]
```diff
@@ -1037,17 +1037,16 @@ func (c *client) processRemoteSub(argo []byte, hasOrigin bool) (err error) {
...
-			expire = true
+		// if the option of retrieving accounts later exists, create an expired one.
+		// When a client comes along, expiration will prevent it from being used,
+		// cause a fetch and update the account to what is should be.
+		if staticResolver {
+			c.Errorf("Unknown account %q for remote subject %q", accountName, sub.subject)
+			return
 		}
+		c.Debugf("Unknown account %q for remote subject %q", accountName, sub.subject)
+		expire = true
```
Comment (kozlovic) [inline server/route.go:1048]: No big deal, but expire is initialized to false but now unconditionally set to true, so we could remove `expire` all together (including in the test just below)

**Review by kozlovic — APPROVED** [review]
LGTM, although code could be simplified a bit by removing `expire`...

File: server/route.go (hunk 1 above)
Comment (derekcollison) [reply server/route.go:1048]: Will do.

Comment (derekcollison) [reply server/route.go]: Fixed.

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**